Modernize Go code

vvoland opened the pull request:
- supersedes: https://github.com/moby/moby/pull/51663

Minimum Go version we support is 1.24 now so we can make use of some of newer features.

Used https://pkg.go.dev/golang.org/x/tools/go/analysis/passes/modernize

---


### pkg/plugins: Use `any`



### modernize: Use `b.Loop` (introduced in Go 1.24)



### modernize: Prefer `strings.SplitSeq` instead of `Split`

  Avoids extra allocations. Added in Go 1.24.



### modernize: Use strings.Cut instead of strings.Index where possible



### modernize: Use maps.Copy instead of for loops



### modernize: Use fmt.Appendf

  Added in Go 1.19



### modernize: Use `min` built-in



### modernize: Use range int

  Added in Go 1.22



### reflect: Use `TypeFor` instead of `TypeOf`

  Added in Go 1.22



### modernize: Use slices.Contains



### modernize: Use strings.Builder instead of string concatenation



### modernize: Use `strings.CutSuffix`

  Added in Go 1.20



### daemon/cluster/test: Use `t.Context()`

vvoland committed:
modernize: Use `b.Loop` (introduced in Go 1.24)

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

vvoland committed:
modernize: Prefer `strings.SplitSeq` instead of `Split`

Avoids extra allocations. Added in Go 1.24.

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

vvoland committed:
modernize: Use strings.Cut instead of strings.Index where possible

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

vvoland committed:
modernize: Use maps.Copy instead of for loops

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

vvoland committed:
modernize: Use fmt.Appendf

Added in Go 1.19

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

vvoland committed:
modernize: Use `min` built-in

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

vvoland committed:
modernize: Use range int

Added in Go 1.22

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

vvoland committed:
reflect: Use `TypeFor` instead of `TypeOf`

Added in Go 1.22

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

vvoland committed:
modernize: Use slices.Contains

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

vvoland committed:
modernize: Use strings.Builder instead of string concatenation

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

vvoland committed:
modernize: Use `strings.CutSuffix`

Added in Go 1.20

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

vvoland committed:
daemon/cluster/test: Use `t.Context()`

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

thaJeztah commented on daemon/libnetwork/iptables/firewalld.go:
```diff
@@ -366,10 +365,5 @@ type interfaceNotFound struct{ error }
 func (interfaceNotFound) NotFound() {}
 
 func contains(list []string, val string) bool {
-	for _, v := range list {
```
For some of these, we should look if we can remove the wrapper / abstraction as well (in a follow-up), as `slices.Contains` is probably equally clear on intent

thaJeztah commented on daemon/libnetwork/networkdb/cluster.go:
Can probably get rid of the `found` variable now;

```go
if !slices.Contains(completed, nid) {
	updatedNetworks = append(updatedNetworks, nid)
}
```

thaJeztah reviewed (approved):
overall LGTM - left some comments

vvoland committed:
daemon: Simplify slices.Contains usage

Remove unnecessary intermediate variables and helper functions when
using slices.Contains.

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

vvoland commented on daemon/libnetwork/networkdb/delegate.go:
```diff
@@ -154,11 +155,8 @@ func (nDB *NetworkDB) handleTableEvent(tEvent *TableEvent, isBulkSync bool) bool
+		nodePresent = true
 	}
 
 	if !ok || network.leaving || !nodePresent {
```
For such complex `if` I think it's still good to leave an intermediate variable as the intent is more readable.

`nodePresent` vs "check if a slice does NOT contain something"

thaJeztah reviewed (approved):
LGTM

thaJeztah reviewed (approved):
LGTM
